examples/custom-client-cxx/cli.cpp

```diff
@@ -0,0 +1,232 @@
+
+  std::vector<boost::filesystem::path> cfg_dirs;
+  auto env{boost::this_process::environment()};
+  if (env.end() != env.find("AKLITE_CONFIG_DIR")) {
+    cfg_dirs.emplace_back(env.get("AKLITE_CONFIG_DIR"));
+  } else {
+    cfg_dirs = AkliteClient::CONFIG_DIRS;
+  }
+
+  if (!local_repo_path.empty()) {
+    local_update_source = {local_repo_path + "/tuf", local_repo_path + "/ostree_repo", local_repo_path + "/apps"};
+    is_local_update = true;
+    LOG_INFO << "Offline mode. Updates path=" << local_repo_path;
+  } else {
+    LOG_INFO << "Online mode";
+  }
+
+  std::unique_ptr<AkliteClient> client;
+  try {
+    client = std::make_unique<AkliteClient>(cfg_dirs);
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "Failed to initialize the client: " << exc.what();
+    return EXIT_FAILURE;
+  }
+
+  auto status = aklite::cli::Pull(*client, -1, target_name, true, is_local_update ? &local_update_source : nullptr);
+  printStatus(status);
+  return aklite::cli::IsSuccessCode(status) ? EXIT_SUCCESS : EXIT_FAILURE;
+}
+
+int cmd_install(std::string target_name, std::string local_repo_path) {
+  boost::log::core::get()->set_filter(boost::log::trivial::severity >= boost::log::trivial::info);
+  bool is_local_update{false};
+  LocalUpdateSource local_update_source;
+
+  std::vector<boost::filesystem::path> cfg_dirs;
+  auto env{boost::this_process::environment()};
+  if (env.end() != env.find("AKLITE_CONFIG_DIR")) {
+    cfg_dirs.emplace_back(env.get("AKLITE_CONFIG_DIR"));
+  } else {
+    cfg_dirs = AkliteClient::CONFIG_DIRS;
+  }
+
+  if (!local_repo_path.empty()) {
+    local_update_source = {local_repo_path + "/tuf", local_repo_path + "/ostree_repo", local_repo_path + "/apps"};
+    is_local_update = true;
+    LOG_INFO << "Offline mode. Updates path=" << local_repo_path;
+  } else {
+    LOG_INFO << "Online mode";
+  }
+
+  std::unique_ptr<AkliteClient> client;
+  try {
+    client = std::make_unique<AkliteClient>(cfg_dirs);
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "Failed to initialize the client: " << exc.what();
+    return EXIT_FAILURE;
+  }
+
+  auto status = aklite::cli::Install(*client, -1, target_name, InstallMode::All, true,
+                                     is_local_update ? &local_update_source : nullptr, aklite::cli::PullMode::None);
+  printStatus(status);
+  return aklite::cli::IsSuccessCode(status) ? EXIT_SUCCESS : EXIT_FAILURE;
+}
+
+int cmd_run() {
+  boost::log::core::get()->set_filter(boost::log::trivial::severity >= boost::log::trivial::info);
+  bool is_local_update{false};
+  LocalUpdateSource local_update_source;
+
+  std::vector<boost::filesystem::path> cfg_dirs;
+  auto env{boost::this_process::environment()};
+  if (env.end() != env.find("AKLITE_CONFIG_DIR")) {
+    cfg_dirs.emplace_back(env.get("AKLITE_CONFIG_DIR"));
+  } else {
+    cfg_dirs = AkliteClient::CONFIG_DIRS;
+  }
+
+  std::unique_ptr<AkliteClient> client;
+  try {
+    client = std::make_unique<AkliteClient>(cfg_dirs);
```

Comment:
The `run` command reports failure while it is actually succeeds in the "happy path" flow.
```
[+] Building 0.0s (0/0)                                                                                                                                      docker:default
[2024-02-06 06:00:23.284611] [0x00007f507d90a340] [info]    Finalization has been completed successfully; Target: intel-corei7-64-lmp-30
[2024-02-06 06:00:23.292628] [0x00007f507d90a340] [info]    Use existing SQL storage: "/work/fio/prjs/lmp-device-register/fioed-device/sql.db"
[2024-02-06 06:00:23.296813] [0x00007f507d90a340] [error]   There is no pending installation to complete
FAILURE: No pending installation to run

```
The problem is that the `AkliteClient` object is instantiated with `finalize=true` while it should be `false` so the finalization/run can be executed and processed in `aklite::cli::CompleteInstall` command.

The same is for the `AkliteClient` instantiation for the rest commands.

Reply:
Updated all `AkliteClient` instantiations, setting `finalize=false`.